dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/get-fixed-fee-for-facility.ts

```diff
@@ -1,58 +1,22 @@
 import { ReportPeriod } from '@ukef/dtfs2-common';
-import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
-import { NotFoundError } from '../../../../../errors';
-import { convertTimestampToDate, getLatestCompletedAmendmentCoverEndDate } from '../../../../../helpers';
+import { calculateUkefShareOfUtilisation, getLatestTfmFacilityValues } from '../../../../../helpers';
 import { calculateFixedFee } from './calculate-fixed-fee';
 
-/**
- * Gets the latest values for the TFM facility with the supplied facility id
- * @param facilityId - The facility id
- * @returns The latest values
- */
-const getLatestTfmFacilityValues = async (
-  facilityId: string,
-): Promise<{
-  coverEndDate: Date;
-  coverStartDate: Date;
-  dayCountBasis: number;
-  interestPercentage: number;
-}> => {
-  const tfmFacility = await TfmFacilitiesRepo.findOneByUkefFacilityId(facilityId);
-  if (!tfmFacility) {
-    throw new NotFoundError(`Failed to find a tfm facility with ukef facility id '${facilityId}'`);
-  }
-
-  const { coverEndDate: snapshotCoverEndDate, coverStartDate, dayCountBasis, interestPercentage } = tfmFacility.facilitySnapshot;
-  const latestAmendedCoverEndDate = getLatestCompletedAmendmentCoverEndDate(tfmFacility);
-
-  const coverEndDate = latestAmendedCoverEndDate ?? snapshotCoverEndDate;
-  if (!coverEndDate) {
-    throw new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`);
-  }
-
-  if (!coverStartDate) {
-    throw new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`);
-  }
-
-  return {
-    coverEndDate: convertTimestampToDate(coverEndDate),
-    coverStartDate: convertTimestampToDate(coverStartDate),
-    dayCountBasis,
-    interestPercentage,
-  };
-};
-
 /**
  * Gets the fixed fee for the given report period
+ * calculates ukef share of utilisation from provided utilisation value and cover percentage
  * @param facilityId - The (ukef) facility id
  * @param utilisation - The facility utilisation
  * @param reportPeriod - The report period
  * @returns The fixed fee for the supplied report period
  */
 export const getFixedFeeForFacility = async (facilityId: string, utilisation: number, reportPeriod: ReportPeriod) => {
-  const { coverEndDate, dayCountBasis, interestPercentage } = await getLatestTfmFacilityValues(facilityId);
+  const { coverEndDate, dayCountBasis, interestPercentage, coverPercentage } = await getLatestTfmFacilityValues(facilityId);
+
+  const calculatedUtilisation = calculateUkefShareOfUtilisation(utilisation, coverPercentage);
+
   return calculateFixedFee({
-    utilisation,
+    utilisation: calculatedUtilisation,
```

Comment:
Please could we rename the `calculateFixedFee` param to be something like `ukefShareOfUtilisation` to make it clear we expect the conversion to happen before calling `calculateFixedFee`?